src/cargo/core/resolver/resolve.rs

```diff
@@ -97,6 +98,21 @@ impl ResolveVersion {
     pub fn max_stable() -> ResolveVersion {
         ResolveVersion::V3
     }
+
+    /// Gets the default lockfile version for the given Rust version.
+    pub fn with_rust_version(rust_version: Option<&RustVersion>) -> Self {
+        let Some(rust_version) = rust_version else {
+            return ResolveVersion::default();
```

Comment:
Should we remove `Default` impl to help force people to use `with_rust_version`?

Reply:
A even terrible idea: we rename it to `fn default(…)` so people never misuse 🙈.

Reply:
addressed this in an awkward way but easier to refer to: <https://github.com/rust-lang/cargo/pull/12861/commits/991663ca614e1f7e48541d632d71766a17282490>. Let me know if there are other strategies to communicate this.